### PR TITLE
Fix xmlrpc tests

### LINKED ISSUE #113: Rescue tests
We need to rescue the tests in each app.

- simplify the fixtures. Create required objects on-demand instead of using a JSON file to provide fixture data.
- Fix existent tests so that they are able to run. Rewrite some of them if necessary.

- [x] testplans
- [x] testcases
- [x] testruns
- [x] core
- [x] xmlrpc

### PR DESCRIPTION
**Pull request by tkdchen**
Fix #113

Signed-off-by: Chenxiong Qi <qcxhome@gmail.com>

### CODE REVIEW

File: tcms/xmlrpc/tests/test_backward_compatibility.py [hunk 1]
```diff
@@ -1,100 +0,0 @@
...
-
-    alias_fields = serializer_class.extra_fields['alias']
-    for original_name, mapped_name in alias_fields.iteritems():
-        del serialize_result[mapped_name]
-
-
-class TestSerializationBackwardCompatibility(FastFixtureTestCase):
-    '''Ensure new serialization method to generate same data with original
-
-    Remove this test when migrate to new serialization completely.
```
Comment (atodorov) [inline tcms/xmlrpc/tests/test_backward_compatibility.py]: Hmmm, do we not want to test with existing data ? Can you explain more about this particular file and why it was removed ?

File: tcms/xmlrpc/tests/test_testcase.py [hunk 2]
```diff
@@ -1,76 +1 @@
 # -*- coding: utf-8 -*-
-
-import unittest
-
-from tcms.testcases.models import TestCase
-
-
-class TestTestCaseSerialization(unittest.TestCase):
```
Comment (atodorov) [inline tcms/xmlrpc/tests/test_testcase.py]: Same question as before, why do we need to delete this?

File: tcms/xmlrpc/tests/test_testrun.py [hunk 3]
```diff
@@ -1,69 +1 @@
 # -*- coding: utf-8 -*-
-
-import unittest
-
-from tcms.testruns.models import TestRun
-
-
-class TestTestRun(unittest.TestCase):
-
```
Comment (atodorov) [inline tcms/xmlrpc/tests/test_testrun.py]: Again, why delete ?

File: tcms/xmlrpc/tests/test_backward_compatibility.py (hunk 1 above)
Comment (tkdchen) [inline tcms/xmlrpc/tests/test_backward_compatibility.py]: This test case was written for ensuring new serialization classes in ``xmlrpc/serializer.py`` work as expected and the API returns same serialized data as before (before the change). This test case should had been removed long time ago, but it's still there as of this PR. So, remove it.

File: tcms/xmlrpc/tests/test_testcase.py (hunk 2 above)
Comment (tkdchen) [inline tcms/xmlrpc/tests/test_testcase.py]: serializer is tested in ``test_serializer.py``.

File: tcms/xmlrpc/tests/test_testrun.py (hunk 3 above)
Comment (tkdchen) [inline tcms/xmlrpc/tests/test_testrun.py]: Same as above.

**Pull request merged**